### PR TITLE
Require Dart 3.0, update and fix lints

### PR DESCRIPTION
**Pull request by kevmoo**

### CODE REVIEW

File: pubspec.yaml [hunk 1]
```diff
@@ -1,19 +1,19 @@
...
+version: 1.0.1-wip
 description: A library for parsing and analyzing CSS (Cascading Style Sheets).
 repository: https://github.com/dart-lang/csslib
 
 topics:
   - css
 
 environment:
-  sdk: '>=2.19.0 <4.0.0'
+  sdk: ^3.0.0
```
Comment (natebosch) [inline pubspec.yaml:10]: We should audit to see if we need any `final`, `interface`, or `base` classes while we are bumping the  version.

Comment (kevmoo) [reply pubspec.yaml:10]: I got no warnings 🤷

Comment (natebosch) [reply pubspec.yaml:10]: Right, there are no warnings about being maximally permissive, but if any of the classes are explicitly not designed for extension or implementation then we'd be sending the wrong signal leaving everything at the default without usage restrictions.

Comment (kevmoo) [reply pubspec.yaml:10]: I don't *think* we're super worried about folks extending these types, right?

Comment (natebosch) [reply pubspec.yaml:10]: I think the best practice would be to consider class modifiers with the sdk bump, but I agree it's probably ok to take a shortcut for this package.

**Review by natebosch — APPROVED** [review]

**Pull request merged**